### PR TITLE
gh-126012: Add `__class_getitem__` to `memoryview`

### PR DESCRIPTION
**Pull request by brianschubert**
* Issue: gh-126012

----
📚 Documentation preview 📚: https://cpython-previews--126013.org.readthedocs.build/

### CODE REVIEW

File: Doc/whatsnew/3.14.rst [hunk 1]
```diff
@@ -194,6 +194,9 @@ Other language changes
   :mod:`copyable <copy>`.
   (Contributed by Serhiy Storchaka in :gh:`125767`.)
 
+* The :class:`memoryview` type now supports subscripting, making it a
+  :term:`generic type`. (Contributed by Brian Schubert in :gh:`126012`.)
```
Comment (picnixz) [inline Doc/whatsnew/3.14.rst]:
```suggestion
* The :class:`memoryview` type now supports subscription,
  making it a :term:`generic type`.
  (Contributed by Brian Schubert in :gh:`126012`.)
```

File: Misc/NEWS.d/next/Core and Builtins/2024-10-26-13-32-48.gh-issue-126012.2KalhG.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+The :class:`memoryview` type now supports subscripting, making it a
```
Comment (picnixz) [inline Misc/NEWS.d/next/Core and Builtins/2024-10-26-13-32-48.gh-issue-126012.2KalhG.rst]:
```suggestion
The :class:`memoryview` type now supports subscription, making it a
```

**Review by picnixz — COMMENTED** [review]
Small nitpicks. We should also add some `versionchanged` for memoryview objects saying that they are now generics.

**Review by picnixz — APPROVED** [review]
Looks good!

File: Doc/library/stdtypes.rst [hunk 3]
```diff
@@ -3889,6 +3889,9 @@ copying.
    .. versionchanged:: 3.5
       memoryviews can now be indexed with tuple of integers.
 
+   .. versionchanged:: 3.14
```
Comment (ZeroIntensity) [inline Doc/library/stdtypes.rst]: *Slowly trying to get people to use this*
```suggestion
   .. versionchanged:: next
```

**Review by ZeroIntensity — APPROVED** [review]
Might as well hop on the approval train :)

**Review by rruuaanng — APPROVED** [review]
LGTM

**Review by sobolevn — APPROVED** [review]
Thank you!

**Review by furkanonder — APPROVED** [review]

**Pull request merged**